cabal.project

```diff
@@ -25,6 +25,8 @@ packages:
          ./plugins/hls-ormolu-plugin
          ./plugins/hls-call-hierarchy-plugin
 
+optional-packages: vendored/*/*.cabal
+
```

Comment:
what's this for?

Reply:
Being able to use vendored packages. If they don't exist, nothing happens. If they exist, they get picked up.

This is required because the network package is broken on M1 and needs a autoreconf (but cabal is unable to do so for you). So we do it manually and put it into this directory.

Reply:
it probably deserves a comment "Used by the M1 build script"

Reply:
hmm it is a somewhat standard way to include potential optional packages in any situation but a comment with an actual use case would be fine

Reply:
If it's (evidently) not obvious to the most active maintainers of this project, it definitely needs a comment for the rest of us poor mortals :)

Reply:
The comment could even point to the script where it's used!

Reply:
You know, github has a feature to add code suggestions ;)

Reply:
@hasufell i hope you dont mind i've added a pair of comments directly in your branch